music_assistant/controllers/player_queues.py

```diff
@@ -1040,6 +1041,23 @@ async def _load_item(
                 and queue_item.media_item.album.item_id == next_item.media_item.album.item_id
             )
         )
+        current_index = self._queues[queue_id].current_index
+        previous_track_from_same_album = (
+            current_index is not None
+            and (previous_index := max(current_index - 1, 0))
```

Comment:
ah, this is exactly what I meant above in my comment. Just delete the line 1044 then